### PR TITLE
When nodeport is enabled, use strict kube-proxy-replacement to ensure…

### PR DESCRIPTION
**Pull request by olemarkus**
… cilium fully replaces kube-proxy

### CODE REVIEW

**Comment by olemarkus** [comment]
Without this, service forwarding fails on newer Cilium versions.

/bug

**Comment by diegocn** [comment]
- Tested @olemarkus's branch and I was able to spin up a running cluster with cilium 1.7.0 and without kube-proxy. 
- I can see some logs on cilium-agents related to some sort of failure on BFP probe even with the host/node AMI with BPF enabled and mounted:
`level=warning msg="BPF system config check: NOT OK." error="CONFIG_BPF kernel parameter is required" subsys=linux-datapath`
- Before this patch kops-controller was not able to reach kubernetes-api. Now it's running as expected.

**Comment by olemarkus** [comment]
See https://github.com/cilium/cilium/issues/10256 regarding the config check error. It is harmless

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -121,6 +121,9 @@ data:
   install-iptables-rules: "{{- if .IPTablesRulesNoinstall -}}false{{- else -}}true{{- end -}}"
   auto-direct-node-routes: "{{- if .AutoDirectNodeRoutes -}}true{{- else -}}false{{- end -}}"
   enable-node-port: "{{- if .EnableNodePort -}}true{{- else -}}false{{- end -}}"
+  {{ if .EnableNodePort }}
+  kube-proxy-replacement: strict
+  {{ end }}
```
Comment (hakman) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12.yaml.template]: Should be easier to deal with the extra white space:
```suggestion
  kube-proxy-replacement: "{{- if .EnableNodePort -}}strict{{- else -}}probe{{- end -}}"
```

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template [hunk 2]
```diff
@@ -121,6 +121,9 @@ data:
   install-iptables-rules: "{{- if .IPTablesRulesNoinstall -}}false{{- else -}}true{{- end -}}"
   auto-direct-node-routes: "{{- if .AutoDirectNodeRoutes -}}true{{- else -}}false{{- end -}}"
   enable-node-port: "{{- if .EnableNodePort -}}true{{- else -}}false{{- end -}}"
+  {{ if .EnableNodePort }}
+  kube-proxy-replacement: strict
+  {{ end }}
```
Comment (hakman) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template]:
```suggestion
  kube-proxy-replacement: "{{- if .EnableNodePort -}}strict{{- else -}}probe{{- end -}}"
```

Comment (olemarkus) [reply upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template]: I would perhaps go for `kube-proxy-replacement: disabled`. it is a bit easier to understand what is happening if one is explicit.

Comment (hakman) [reply upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template]: Sounds even better to me.

**Comment by johngmyers** [comment]
@olemarkus This looks like the prime suspect in breaking the [kops-aws-cni-cilium](https://testgrid.k8s.io/sig-cluster-lifecycle-kops#kops-aws-cni-cilium) e2e test. Could you take a look?

**Comment by olemarkus** [comment]
yep

**Pull request merged**